apps/emqx/src/emqx_schema.erl

```diff
@@ -941,6 +945,65 @@ fields("sysmon_os") ->
       }
     ];
 
+fields("sysmon_top") ->
+    [ {"num_items",
+       sc(non_neg_integer(),
+         #{ mapping => "system_monitor.top_num_items"
+          , default => 10
+          , desc => "The number of top processes per monitoring group"
+          })
+      }
+    , {"sample_interval",
+       sc(emqx_schema:duration(),
+         #{ mapping => "system_monitor.top_sample_interval"
+          , default => "2s"
+          , desc => "Specifies how often process top should be collected"
+          })
+      }
+    , {"max_procs",
+       sc(non_neg_integer(),
+         #{ mapping => "system_monitor.top_max_procs"
+          , default => 3000000
```

Comment:
Q: Could this default be too high?  I've noticed that it may take a bit of memory when there are a lot of processes and this is high.

Reply:
Good question. System monitor is disabled by default (when sink is not set), and I assume that the users who enable are conscious of the performance impact.